lambdas/data-migration2/tests/test-pdrs.js

```diff
@@ -0,0 +1,344 @@
+      pan_message: null,
+      stats: null,
+      address: null,
+      original_url: null,
+      timestamp: null,
+      duration: null,
+      created_at: new Date(testPdr.createdAt),
+    }
+  );
+});
+
+test.serial('migratePdrRecord throws RecordAlreadyMigrated error for already migrated record', async (t) => {
+  const { knex, testCollection, testProvider } = t.context;
+
+  const testPdr = generateTestPdr(testCollection.name, testProvider.name);
+  await migratePdrRecord(testPdr, knex);
+
+  await t.throwsAsync(
+    migratePdrRecord(testPdr, knex),
+    { instanceOf: RecordAlreadyMigrated }
+  );
+});
+
+test.serial('migratePdrs skips already migrated record', async (t) => {
+  const {
+    knex,
+    testCollection,
+    testProvider,
+  } = t.context;
+  const testPdr = generateTestPdr(testCollection.name, testProvider.name);
+
+  await migratePdrRecord(testPdr, knex);
+  await pdrsModel.create(testPdr);
+
+  const migrationSummary = await migratePdrs(process.env, knex);
+  t.deepEqual(migrationSummary,
+    {
+      dynamoRecords: 1,
+      skipped: 1,
+      failed: 0,
+      success: 0,
+    });
+
+  const records = await knex(tableNames.pdrs).where({ name: testPdr.pdrName });
+  t.is(records.length, 1);
+  t.teardown(() => pdrsModel.delete({ pdrName: testPdr.pdrName }));
+});
+
+test.serial('migratePdrs processes multiple PDR records', async (t) => {
+  const {
+    collectionPgModel,
+    knex,
+    providerPgModel,
+    testCollection,
+    testProvider,
+  } = t.context;
+
+  const fakeCollection = fakeCollectionRecordFactory();
+  const fakeProvider = fakeProviderRecordFactory();
+  await collectionPgModel.create(knex, fakeCollection);
+  await providerPgModel.create(knex, fakeProvider);
+
+  const testPdr = generateTestPdr(testCollection.name, testProvider.name);
+  const anotherPdr = generateTestPdr(fakeCollection.name, fakeProvider.name);
+
+  await Promise.all([
+    pdrsModel.create(testPdr),
+    pdrsModel.create(anotherPdr),
+  ]);
+  t.teardown(() => Promise.all([
+    pdrsModel.delete({ pdrName: testPdr.pdrName }),
+    pdrsModel.delete({ pdrName: anotherPdr.pdrName }),
+  ]));
+  const migrationSummary = await migratePdrs(process.env, knex);
+  t.deepEqual(migrationSummary, {
+    dynamoRecords: 2,
+    skipped: 0,
+    failed: 0,
+    success: 2,
+  });
+  const records = await knex(tableNames.pdrs);
+  t.is(records.length, 2);
+});
+
+test.serial('migratePdrs processes all non-failing records', async (t) => {
+  const {
+    collectionPgModel,
+    knex,
+    providerPgModel,
+    testCollection,
+    testProvider,
+  } = t.context;
+
+  const fakeCollection = fakeCollectionRecordFactory();
+  const fakeProvider = fakeProviderRecordFactory();
+  await collectionPgModel.create(knex, fakeCollection);
+  await providerPgModel.create(knex, fakeProvider);
+
+  const testPdr = generateTestPdr(testCollection.name, testProvider.name);
+  const anotherPdr = generateTestPdr(fakeCollection.name, fakeProvider.name);
```

Comment:
Same comment: I think you can just re-use the `testCollection` and `testProvider` here